doc/api/assert.md

```diff
@@ -2548,6 +2548,232 @@ assert.throws(throwingFirst, /Second$/);
 Due to the confusing error-prone notation, avoid a string as the second
 argument.
 
+## `assert.partialDeepEqual(actual, expected[, message])`
```

Comment:
The existence of the `strict` method is something that is very confusing and I strongly recommend we do not add any "non"-strict version. I could not find anyone asking for this either.

That should never be used. If we ever want to have a different comparison, it should be done by e.g., `assert.any(Number)` or similar.

Reply:
I give you a whole discussion around wanting this https://github.com/fastify/fastify/discussions/5628

Reply:
@jsumners I am absolutely +1 on adding some kind of partial deep equal comparison as done here. This PR just adds three different APIs while I believe we should only ship the `partialDeepStrictEqual` one. I am not a super fan of the name but I know there was a lot of back and forth and it is tricky to find the right name.

@puskin94 what about doing the following as next steps:

1. Remove everything besides the current `partialDeepStrictEqual()` implementation.
2. Have a small poll for the name.

That way we could ship the main requested feature and later on decide if we want to add anything else on top.

I think this is great work and we should definitely ship that functionality!